### PR TITLE
Use Redis for caching in test environment

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

All other environments use it. Tests will be more realistic when using Redis for caching. We are also more likely to see bugs in redis-related gems.

This came up when looking into flaky specs but it doesn't actually solve any current issue. Still thought it would be nice to have.

#### What should we test?

- No manual test. This affects only CI and local test environments.

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by abdellani — APPROVED** [review]

**Review by jibees — APPROVED** [review]
I agree! 

Shouldn't we remove that `caching: true` option used in several places as well and defined like this:

https://github.com/openfoodfoundation/openfoodnetwork/blob/master/spec/base_spec_helper.rb#L94-L101

File: config/environments/test.rb [hunk 1]
```diff
@@ -14,7 +14,11 @@
   config.public_file_server.headers = { 'Cache-Control' => 'public, max-age=3600' }
 
   # Separate cache stores when running in parallel
-  config.cache_store = :file_store, Rails.root.join("tmp", "cache", "paralleltests#{ENV['TEST_ENV_NUMBER']}")
```
Comment (Matt-Yorkley) [inline config/environments/test.rb:17]: Was this line previously guarding against parallelisation issues?

I think we need to account for:
- leaking between different parallel tests in CI
- leaking between dev and test environments locally (dev and test should not share the same cache space)

For starters we could at least change `redis://localhost:6379/1` to `redis://localhost:6379/2` here, which I think would solve the second issue.

Comment (dacook) [reply config/environments/test.rb:17]: Good pickup! It was, but it looks like we haven't used local parallelisation for a long time:
(TEST_ENV_NUMBER is a variable for the parallel_tests gem which has been removed in 4cbc449a51ce4836152984938e1f161976b6a8fe)
On CI, tests are run in separate environments so the first point is safe.

For the second point, I think that suggestion is a very good idea 👍

**Review by dacook — CHANGES_REQUESTED** [review]
I agree with Matt about using a Redis database separate to dev, will try changing it.

File: config/environments/test.rb [hunk 2]
```diff
@@ -14,7 +14,11 @@
...
+  config.cache_store = :redis_cache_store, {
+    driver: :hiredis,
+    url: ENV.fetch("OFN_REDIS_URL", "redis://localhost:6379/1"),
```
Comment (dacook) [inline config/environments/test.rb:19]:
```suggestion
    url: ENV.fetch("OFN_REDIS_URL", "redis://localhost:6379/2"),
```

**Comment by dacook** [comment]
Manual test: I didn't test for cache leakage, but at least tested the test suite works locally. 
It mostly works (about as much as it usually would I think..)

**Review by dacook — APPROVED** [review]

**Review by mkllnk — COMMENTED** [review]
Thank you!

**Pull request merged**